src/main/java/frc/robot/commandGroups/MidScoring.java

```diff
@@ -0,0 +1,27 @@
+package frc.robot.commandGroups;
+
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.subsystems.arm.ArmConstants;
+import frc.robot.subsystems.arm.commands.SetArmsPosition;
+import frc.robot.subsystems.gripper.Gripper;
+import frc.robot.subsystems.vision.Limelight;
+import frc.robot.subsystems.vision.VisionConstants;
+
+public class MidScoring extends SequentialCommandGroup {
+    private final Limelight limelight = Limelight.getInstance();
+    private final Gripper gripper = Gripper.getInstance();
+    public MidScoring(){
+        addCommands(
+                new ConditionalCommand(
+                        new TapeCommandGroup(),
+                        new AprilTagCommandGroup(),
+                        () -> limelight.getPipeline() == VisionConstants.REFLECTIVE_TAPE_PIPELINE
+                ),
+                new SetArmsPosition(() -> limelight.getPipeline() == VisionConstants.REFLECTIVE_TAPE_PIPELINE ? ArmConstants.MIDDLE_CONE_SCORING : ArmConstants.MIDDLE_CUBE_SCORING),
```

Comment:
Citing from the [Planning Meeting Summary](https://smooth-macaroon-4c5.notion.site/Planning-Meeting-2c65c00b456040449f5154aeab):

> Scoring command
> 1. Switch pipeline to April Tags pipeline.
> 2. Start driving autonomously
>     1. to the Cube Node, if holding a Cube.
>     2. to the Cone Node, if holding a Cone.

So choosing the arm position can be simpler.
How is the Limelight  pipeline chosen?

Reply:
IIRC the driver will have a button to change the pipeline
and I'm pretty sure we agreed that the driver will drive manually to the node they want to put the game piece in and than the robot will adjust itself and score the game piece

Reply:
So IIUC, the driver (operator?) chooses the game piece, clicks a button to select the pipeline, then the arm chooses its position based on the pipeline.
Could you choose the position directly based on the desired game piece?
> and I'm pretty sure we agreed that the driver will drive manually

That's not what was written in the meeting summary...

Also, another concern I have is that the shortest distance between two Cone Nodes is only 1.10m. How close does the robot need to be to the Grid line to see only one piece of reflective tape? How will the driver know the robot is not too far away?